pkg/injector/pod_patch.go

```diff
@@ -593,6 +593,14 @@ func getSidecarContainer(annotations map[string]string, id, daprSidecarImage, im
 				Name:  "NAMESPACE",
 				Value: namespace,
 			},
+			{
+				Name: "POD_NAME",
```

Comment:
We need to find another way to either pass the pod name to the container or identify the pod in the operator. This uses the FieldPath ref (Downward API) which is not supported on platforms like Virtual Kubelet.